### ISSUE TITLE
Can NOT run on CentOS 7: /lib64/libc.so.6: version `GLIBC_2.18' not found

### ISSUE THREAD

**Issue by iredmail** [meilisearch/MeiliSearch#1076]
**Describe the bug**

Still can not run the latest 0.16.0 on CentOS 7.8.2003. The error is:

```
meilisearch: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /opt/meilisearch/bin/meilisearch)
```

**To Reproduce**

- Download the latest 0.16.0, file `meilisearch-linux-amd64`
- Setup a new CentOS 7 VM.
- Run it:

```
# ./meilisearch-linux-amd64
./meilisearch: /lib64/libc.so.6: version `GLIBC_2.18' not found (required by ./meilisearch)
```

**Expected behavior**

Running and listening on default port 7700.

**Comment by iredmail** [comment]
#932 is related, but no solution.

**Comment by curquiza** [comment]
Hello @iredmail!

I've just tested and indeed it does not work on CentOS v7.
However, it works on CentOS v8.

If you cannot upgrade your CentOS version you can use docker or compile from the sources.
See this installation guide: https://docs.meilisearch.com/guides/advanced_guides/installation.html#installation
(the other solutions I suggest are in the tabs)

**Comment by iredmail** [comment]
We’re looking for running Meilisearch on CentOS 7+8, Debian 9+10, Ubuntu 18.04+20.04, all are fine except CentOS 7. 

No problem to use CentOS 8, but it would be great to have it running on CentOS 7.

**Comment by curquiza** [comment]
Yes, I understand this is an issue. There are a few alternatives at this moment:
- Compile natively: this is an Open Source repository you can always clone it and compile it in a specific environment
- Use docker: We provide official images and you can use that abstraction layer

I am really sorry but being realistic, even if we would like to, I don't think we can prioritize supporting CentOS 7 at this moment. If you ever fix the problem in a different way, we would be glad to have any contribution that would let MeiliSearch support this natively 🙂

**Comment by MarinPostma** [comment]
Hello @iredmail,

if I'm not wrong, centos7 libc version is 2.17 (you can check that by running `ldd --version`), meilisearch is built on ubuntu whose libc version is 2.18, hence the mismatch in version.

If you really want to run meilisearch on centos7, the easiest would be to build meilisearch from source on centos7:

```bash
# install rust
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
# install gcc
sudo yum install gcc
# download meilisearch sources
curl -L https://github.com/meilisearch/MeiliSearch/archive/v0.16.0.tar.gz > meilisearch .tar.gz
# build rust
~/.cargo/bin/cargo build --release
```

**Comment by iredmail** [comment]
Thanks for helping. :)

No problem to build it with source code, my purpose was raising this issue and maybe you guys want to figure out a way to offer binary package for CentOS 7 users. like updating CI to build it automatically or something.

**Comment by hexiaoting** [comment]
>cd MeiliSearch-0.16.0 && ~/.cargo/bin/cargo build --release
version 0.16.0 can be build from source on CentoOS 7. But the latest version cannot be build, the error info：
```
error: /lib64/libc.so.6: version `GLIBC_2.29' not found (required by /mnt/disk3/hewenting/download/meilisearch/github-meilisearch/target/release/deps/libpaste-3d7edb92b2fae99d.so)
 --> /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/boxed.rs:6:5
  |
6 | use paste::paste;
  |     ^^^^^

error: cannot determine resolution for the macro `paste`
  --> /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/boxed.rs:15:9
   |
15 |         paste! {
   |         ^^^^^
...
34 | service_object!(BoxService, Box, service);
   | ----------------------------------------- in this macro invocation
   |
   = note: import resolution is stuck, try simplifying macro imports
   = note: this error originates in the macro `service_object` (in Nightly builds, run with -Z macro-backtrace for more info)

error: cannot determine resolution for the macro `paste`
  --> /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/boxed.rs:15:9
   |
15 |         paste! {
   |         ^^^^^
...
35 | service_object!(RcService, Rc, rc_service);
   | ------------------------------------------ in this macro invocation
   |
   = note: import resolution is stuck, try simplifying macro imports
   = note: this error originates in the macro `service_object` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0412]: cannot find type `BoxService` in this scope
  --> /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/boxed.rs:89:19
   |
89 |         Service = BoxService<Req, Res, Err>,
   |                   ^^^^^^^^^^ help: a trait with a similar name exists: `Service`
   |
  ::: /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/lib.rs:93:1
   |
93 | pub trait Service<Req> {
   | ---------------------- similarly named trait `Service` defined here

error[E0412]: cannot find type `BoxService` in this scope
  --> /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/boxed.rs:90:35
   |
90 |         Future = BoxFuture<Result<BoxService<Req, Res, Err>, InitErr>>,
   |                                   ^^^^^^^^^^ help: a trait with a similar name exists: `Service`
   |
  ::: /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/lib.rs:93:1
   |
93 | pub trait Service<Req> {
   | ---------------------- similarly named trait `Service` defined here

error[E0412]: cannot find type `BoxService` in this scope
   --> /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/boxed.rs:105:20
    |
105 |     type Service = BoxService<Req, Res, Err>;
    |                    ^^^^^^^^^^ help: a trait with a similar name exists: `Service`
    |
   ::: /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/lib.rs:93:1
    |
93  | pub trait Service<Req> {
    | ---------------------- similarly named trait `Service` defined here

error[E0412]: cannot find type `BoxService` in this scope
   --> /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/boxed.rs:131:20
    |
131 |     type Service = BoxService<Req, Res, Err>;
    |                    ^^^^^^^^^^ help: a trait with a similar name exists: `Service`
    |
   ::: /home/hewenting/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-service-2.0.2/src/lib.rs:93:1
    |
93  | pub trait Service<Req> {
    | ---------------------- similarly named trait `Service` defined here

   Compiling regex v1.5.6
   Compiling lzma-rs v0.2.0
   Compiling rustc_version v0.4.0
   Compiling brotli v3.3.4
For more information about this error, try `rustc --explain E0412`.
error: could not compile `actix-service` due to 7 previous errors
warning: build failed, waiting for other jobs to finish...
```

**Comment by irevoire** [comment]
Ooops, I don’t think we noticed that!
It probably happened when we bumped all of our dependencies.
@meilisearch/core-team do we want to do something about that?

The glibc_2.18 dated back to 2012 and was supported by default on most distributions, while the new minimum version required; the glibc 2.29 was released in January 2019 (only 3.5 yo).

It’s obviously not available in centOS, but it’s not available either in ubuntu 18.04 or in debian 10 which still have a lot of users 😔
